### ISSUE TITLE
Regression: 4 intersections are found between 2 circles instead of 2

### ISSUE THREAD

**Issue by aecz** [paperjs/paper.js#1638]
# Description/Steps to reproduce

![Image of circle intersections](https://user-images.githubusercontent.com/7781120/51199147-50e95080-18f6-11e9-9b6d-e2822954c593.png)

```js
project.currentStyle = {
    strokeColor:"red"
};

var circle1 = new Path.Circle([100,100],100);
var circle2 = new Path.Circle([150,150],100);

circle1.getIntersections(circle2)
    .forEach(function (i) {
        new Path.Circle(i.point,4);
    });
```
4 intersections are shown. 2 correct and 2 wrong on the first circle.
version <= 0.10.2 : Works
0.12.0 <= version >= 0.10.3 : Fails

# Link to reproduction test-case

http://sketch.paperjs.org/#V/0.11.8/S/bZDBCsIwDIZfpfS0wShO9LLhSTx4EzxaD7VGVzdTyaIisne3nYoiBhJI8vHnJ3eJ5giykMsa2FYyk9ZvY38ifwDLyp6JAHnJtwbERNw1ihAtk69h6htPhZYEWy01dqVGjRdDwjqyDeSBR7iKheFKTftRssoHgyzkOpa0/MaHf/FxwMcfXONLW+2B58hAbXDpPLbJSyV9OlQ7TzNjq2R3xh4QiUvf9mP8nnLq5B1yNopnItClZXjHhsDU/aaVxWrdPQA=

# Expected result

http://sketch.paperjs.org/#V/0.10.2/S/bZDBCsIwDIZfpfS0wShO9LLhSTx4EzxaD7VGVzdTyaIisne3nYoiBhJI8vHnJ3eJ5giykMsa2FYyk9ZvY38ifwDLyp6JAHnJtwbERNw1ihAtk69h6htPhZYEWy01dqVGjRdDwjqyDeSBR7iKheFKTftRssoHgyzkOpa0/MaHf/FxwMcfXONLW+2B58hAbXDpPLbJSyV9OlQ7TzNjq2R3xh4QiUvf9mP8nnLq5B1yNopnItClZXjHhsDU/aaVxWrdPQA=

# Additional information

macOS 10.14
Chrome 73.0 / Firefox 65.0

**Comment by sapics** [comment]
@aecz Thanks for reporting!
It looks bug at addCurveIntersections in Curve.js as [sketch](http://sketch.paperjs.org/#V/prebuilt/S/jZExa8MwEIX/itAkgRGOSRaXTqFDt4LHOIMqn2PVsi6clIRi/N8rG0PSUmgFEtx73zt00si9HoCXvOohmo5n3GAz12fCDzBRmQsR+FjFTwfsmY0hEvawR4dU1pygqfn0VPurJmYsGQebRHm4sTcdO7VfJHHY5HmW9nE+5De8+BXfJXx3x9fOyjgM0KTEmr0LrXYBHkiCAa9QwWlIlxe5/Nsp/u0UD91OEF99BArprSz6INaIVC3SizadaC9+sYSVY+1ZWj/HteqM1sdsO486yfQH7wS6X8TAy8Nx+gI=)
![2019-01-19_085151](https://user-images.githubusercontent.com/11169954/51418573-9bedb700-1bc7-11e9-9be5-8d1679d5ce3d.png)

**Comment by sapics** [comment]
By checking the nearest point is close enough to the other curve, this bug is fixed. https://github.com/sapics/paper.js/commit/df9a751b900e919c7bb1031429b3945d2d75f3ce
But I guess that there is other better way which is not necessary to check the distance.

**Comment by lehni** [comment]
This error was introduced in 4c61153bd4602add6cf8fd5621954b47e84cba2b

**Comment by lehni** [comment]
Changing this:
```js
if (uMax - uMin >= fatLineEpsilon) {
```
to:
```js
if (uMax - uMin === 0 || uMax - uMin >= fatLineEpsilon) {
```
Fixes the issue, while preserving the fix for #1088. But I don't understand why. @iconexperience are you still around? Could you shed some light on this?
